modules/ROOT/pages/apid-behavioral-headers.adoc

```diff
@@ -0,0 +1,98 @@
+====
+
+MS2-Error-Level:: Indicates which range of codes will be considered errors. You can specify one of the following three case-sensitive values:
++
+* `REQUEST_ONLY`: Only 4XX codes are considered to be errors.
+* `SERVER_ONLY`: Only 5XX codes are considered to be errors.
+* `ALL`: Both 4XX and 5XX codes are considered to be error. This is the default value.
+
+MS2-Error-Rate:: Indicates the probability of getting an error status code as a response. The value for this header must be a decimal number between 0 and 1.
++
+This header works together with MS2-Status-Code and MS2-Error-Level, but not both at the same time.
++
+* If you use this header together with MS2-Status-Code and the mocking service (based on the probability) determines that an error code should be returned, the mocking service uses the error code that is specified by MS2-Status-Code. Therefore, for MS2-Status-Code you must specify both a non-error code and an error code, with the two codes separated by a comma.
+* If you use this header together with MS2-Error-Level and the mocking service (based on the probability) determines that an error code should be returned, the mocking service uses one of the error codes that is in the range specified by MS2-Error-Level. If the mocking service determines that a non-error code should be returned, it uses the first non-error code that is defined for the method.
+
+MS2-Randomize:: When this header is sent as true, the mocking service auto-generates a payload response if no example is defined for the response.
++
+For example, suppose that an API specification contained the definitions of this type:
++
+----
+types:
+  Person:
+    properties:
+      id: integer
+      name: string
+----
++
+If you did not provide an example, but you used the header MS2-Randomize, set to `true`, with the mocking service, the payload of the returned message would contain an example with random values that matched the data types, like this:
++
+----
+{
+  "id": 98358,
+  "name": "OVCKr"
+}
+----
++
+Moreover, the header respects more detailed definition of types. For example, you might change your type definition to look like this:
++
+----
+types:
+  Person:
+    properties:
+      id:
+        type: integer
+        minimum: 3
+        maximum: 30
+        multipleOf: 3
+      name: string
+----
++
+If MS2-Randomize were set to `true`, the payload in the response from the mocking service might look like this, with the value of `id` conforming to the constraints:
++
+----
+{
+  "id": 9,
+  "name": "rul68"
+}
+----
+
+MS2-Status-Code:: Specifies the non-error code and the error code that the mocking service can choose from for the return message. Separate the two status codes with a comma. For example, you might want the mocking service to return a 201 message, if there is no error, or a 401 status code, if there is an error; in this case, you would specify `201,401` for the value.
```

Comment:
I thought that you said, for now, the mocking service cannot yet choose codes from the list randomly, but will in a later release. So, for now, the mocking service will only ever choose the first non-error code in this field and the first error code in this field.

Reply:
yes, that's true. But the idea is still the same, MS2-Status-Code is a filter over the codes listed in the spec. 

you said:

> Specifies the non-error code and the error code that the mocking service can choose from for the return message. Separate the two status codes with a comma

According to that, it is as if only two codes can be specified, in fact, one error and one non-error code, when the idea is to filter the codes listed in the spec. For example, You could set only one code  (error or not) and it will be also valid

Reply:
can we say "you specify ..." instead of "you would specify..." 

Not a big deal.

Reply:
@dfmacias , please have a look at my revision. I removed the bullet points and added a table to explain how MS2-Error-Rate works with MS2-Status-Code, MS2-Error-Level, and how it works by itself. 

@dukesphere , that would make the sentence incorrect because I am describing an action that is conditional upon something else happening.